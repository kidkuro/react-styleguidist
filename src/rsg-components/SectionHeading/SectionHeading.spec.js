import React from 'react';
import SectionHeading, { SectionHeadingRenderer, styles } from './SectionHeadingRenderer';

const props = {
	classes: classes(styles),
	id: 'heading',
	href: '/heading',
	depth: 3,
};

it('should pass props to slots', () => {
	const actual = shallow(
		<SectionHeading
			id="heading"
			depth={1}
			href="/heading"
			slotName="slot"
			slotProps={{ foo: 1, bar: 'baz' }}
		>
			Heading
		</SectionHeading>
	);

	expect(actual).toMatchSnapshot();
});

it('renderer should render H4 tag', () => {
	const actual = shallow(<SectionHeadingRenderer {...props}>Heading</SectionHeadingRenderer>);

	expect(actual).toMatchSnapshot();
});

it('renderer should render heading with deprecated styles', () => {
	const actual = shallow(
		<SectionHeadingRenderer {...props} deprecated>
			Heading
		</SectionHeadingRenderer>
	);

	expect(actual).toMatchSnapshot();
});
