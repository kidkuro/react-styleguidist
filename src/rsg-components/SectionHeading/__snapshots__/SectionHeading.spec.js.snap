// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render H4 tag 1`] = `
<h4
  className="root"
  id="heading"
>
  <a
    className="heading heading4"
    href="/heading"
  >
    Heading
  </a>
  <div
    className="toolbar"
  />
</h4>
`;

exports[`renderer should render H6 tag if depth is over 5 1`] = `
<h6
  className="root"
  id="heading"
>
  <a
    className="heading heading6"
    href="/heading"
  >
    Heading
  </a>
  <div
    className="toolbar"
  />
</h6>
`;

exports[`renderer should render heading with deprecated styles 1`] = `
<h4
  className="root"
  id="heading"
>
  <a
    className="heading heading4 isDeprecated"
    href="/heading"
  >
    Heading
  </a>
  <div
    className="toolbar"
  />
</h4>
`;

exports[`should pass props to slots 1`] = `
<SectionHeadingRenderer
  classes={
    Object {
      ".rsg--heading-1:hover, .rsg--heading-1:active": "rsg--heading-1:hover, .rsg--heading-1:active",
      "heading": "rsg--heading-1",
      "heading1": "rsg--heading1-2",
      "heading2": "rsg--heading2-3",
      "heading3": "rsg--heading3-4",
      "heading4": "rsg--heading4-5",
      "heading5": "rsg--heading5-6",
      "heading6": "rsg--heading6-7",
      "isDeprecated": "rsg--isDeprecated-8",
      "root": "rsg--root-0",
      "toolbar": "rsg--toolbar-9",
    }
  }
  depth={1}
  href="/heading"
  id="heading"
  slotName="slot"
  slotProps={
    Object {
      "bar": "baz",
      "foo": 1,
    }
  }
>
  Heading
</SectionHeadingRenderer>
`;
